src/Microsoft.OpenApi/Reader/OpenApiModelFactory.cs

```diff
@@ -279,10 +299,14 @@ private static ReadResult InternalLoad(MemoryStream input, string format, OpenAp
                     var mediaType = response.Content.Headers.ContentType.MediaType;
                     var contentType = mediaType.Split(";".ToCharArray(), StringSplitOptions.RemoveEmptyEntries)[0];
                     format = contentType.Split('/').LastOrDefault();
+                    if (!string.IsNullOrEmpty(format) && format.Contains('-'))
+                    {
+                        format = format.Split('-').LastOrDefault(); // for non-standard MIME types e.g. text/x-yaml used in older libs or apps
+                    }
```

Comment:
```suggestion
                    format = contentType.Split('/').Last().Split('-').Last().Split('+').Last();
                    // for non-standard MIME types e.g. text/x-yaml used in older libs or apps
```

Some context here:
- `string.Empty.Split('-').Last()` returns string empty
- using `-` [is not recommended by the RFC](https://www.rfc-editor.org/rfc/rfc6838#section-3.4)
- the `+` is to support vendor specific media types